### DISCUSSION TITLE
BLE Sensors Development

### DISCUSSION THREAD

**Discussion by HorstvanGrass** [XCSoar/XCSoar#1110] [General]
I recently started some project involving Bluetooth Low Energy (BLE) and asking if there is any development in this direction planed? Since there is a port named so in the code, there was definitely some interest, but I could not find any service writing or reading a char to a server/client.

**Comment by HorstvanGrass** [comment]
I will answer my own question, as I missed some files (https://github.com/XCSoar/XCSoar/blob/master/android/src/BluetoothSensor.java)

Nevertheless, does somebody managed to run android studio to compile the files? The ability to place breakpoint for the task above is crucial. I could not get to work the module correctly in android studio.

**Reply by lordfolken** [reply]
For compiling XCSoar see https://xcsoar.readthedocs.io/en/latest/build.html
Generally we use debian / ubuntu as a compile host. 
We don't really use IDE's unless you consider VIM to be one.

**Reply by HorstvanGrass** [reply]
Compile was never an issue. Actually I prefer to code in an IDE, because I can place breakpoints, monitor variable and see performance losses. For Unix platform this works like a charm using clion. However, as I want to develop my own sensor board using BLE I need the Android side.
I could find a workaround where I can monitor logs from the app while running on a test device. This is not great, but better than be completely blind.

This is tough to develop such a big app completely using vim.

**Reply by lordfolken** [reply]
I usually use LogFormat() to expose variable values on stdout.  For breakpoints etc i use gdb.

**Reply by HorstvanGrass** [reply]
Did you manage to get GDB to work for the android section? That would be so helpful. I'm struggling since at least 2 weeks to get this to work.
What I could archive so far:
- Android is using the native code development (NDK), hence all the C++ files are packed in a shared library. 
- When debugging the app, the process starts but it tell me that libsxcsoar.so does not have debug symbols. Hence no breakpoint possible.
- I checked the makefiles and g++ was used with flag -0g which as far I know should result these debug symbols.
